packages/site/src/routes/[dictionaryId]/entries/print/CitationClipped.svelte

```diff
@@ -0,0 +1,26 @@
+<script lang="ts">
+  export let citation:string;
+  export let citationText:string;
+  const maxLength = 167; //Number based if it looks well on letter format
+  let citationClipped: string,
+    textsSum: number;
+  $: if (citation && citationText) {
```

Comment:
This component's only purpose is to receive names and a maxLength property. Can you please rewrite this PR to instead just add a function with tests that you can use in the +page.svelte file? At present it's very difficult to know what all this logic will do without starting a server and going to pages manually. This sort of thing is best served by a simple function with tests covering each use case. The stripHTMLTags is a good example to learn from. Take the function you have here in this component, put it in a typescript, and add enough tests to cover all the use cases.

Reply:
@jacob-8 That was the first thing I thought: create a function, but how can I get the text inside an HTML element? I could try to add an id to the div element but I don't know when to call the `document.getElementById` function.